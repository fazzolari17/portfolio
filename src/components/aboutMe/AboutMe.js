// import mainImage from '../../images/american-digger.giuseppe-savage.jpg';
import { Link } from 'react-router-dom';
import style from './styleSheet';
import s from '../projects/styleSheet';
import Hero from './Hero';
import Section from './Section';
import Card from '../Card';
import { uppercase } from '../../util/helperFunctions';
import certifications from '../../data/certifications';
import Carousel from '../Carousel';

const AboutMe = () => {
  const numberOfCertifications = 8;

  const renderCertifications = certifications.map(({ certificationUrl, image, altText, id }) => (
    <div style={s.flexCenter}>
      <h2 style={style.certificationTitle}>{uppercase('certifications')}</h2>
      <a style={style.flexCenter} href={certificationUrl} target={'_blank'} rel={'noreferrer'}>
          <img style={style.certificationImg} src={image} alt={altText} />
        </a>
    </div>
  ))

  return (
    <section className='sectionMargin'>
      <Hero />
      <Section invert={true}>
        <Card invert={true} style={{maxWidth: '600px'}}>
        <p>
           {` I'm an avid hiker, biker, backcountry camper, and hanger(Hammocking). I Love all things outdoors, and have a passion for learning. I have an analytical mind and shine when I am problem solving. I am happiest when I am creating something. I have a very diverse career background that has left me with a very unique and large skill set. I have multiple(${numberOfCertifications} at the moment) certifications in tech related fields. My current experience includes REST API, MERN, CRUD, CI/CD, TDD. Plus, I can read the shit out of documentation! `}
          </p>
          <div style={style.listContainer}>
            <div>
              <ul>
                <li>TypeScript</li>
                <li>JavaScript</li>
                <li>Git</li>
                <li>React</li>
                <li>Node</li>
                <li>React Native</li>
                <li>Mongo Db</li>
                <li>Express</li>
           
                <li>SQL</li>
              </ul>
            </div>
            <div>
              <ul>
                <li>HTML5</li>
                <li>CSS3</li>
                <li>Jest</li>
                <li>Mocha</li>
                <li>Cypress E2E</li>
            
              </ul>
            </div>

          </div>
        </Card>
      </Section>
      <Section invert={false}>
          <Carousel slide={renderCertifications} />
      </Section>
    </section>
  )
};

export default AboutMe;