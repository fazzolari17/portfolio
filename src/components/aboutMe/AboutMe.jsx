import React from 'react';
import style from './styleSheet';
// import Carousel from '../Carousel';
import Section from '../Section';
import Card from '../Card';
import Hero from './Hero';
// import { uppercase } from '../../util/helperFunctions';
// import certifications from '../../data/certifications';
import aboutMe from '../../data/aboutme';
// import useViewport from '../../hooks/useViewport';
// import { breakpoint } from '../../constants';
// import MobileCertificates from './MobileCertificates';

const AboutMe = () => {
  // const [isMobile, setIsMobile] = React.useState();
  // const { width } = useViewport();

  // React.useEffect(() => {
  //   if (width > breakpoint) setIsMobile(true);
  //   if (width < breakpoint) setIsMobile(false);
  // }, [width]);


  // const renderCertifications = certifications.map(({ name, certificationUrl, image, altText, id }) => (
  //   <div key={id} style={style.flexCenter}>
  //     <img style={style.certificationImg} src={image} alt={altText} />
  //     <a href={certificationUrl} target={'_blank'} rel={'noreferrer'}>
  //       <p className="legend" style={style.legend}>{uppercase(name)}</p>
  //     </a>
  //   </div>
  // ));


  const renderListOfSkills = () => {
    const half = Math.floor(aboutMe.languages.length / 2);

    const leftList = aboutMe.languages
      .filter((lang, i) => i < half)
      .map((lang, i) => <li key={`${lang}${i}`}>{lang}</li>);

    const rightList = aboutMe.languages
      .filter((lang, i) => i > half)
      .map((lang, i) => (<li key={`${lang}${i}`}>{lang}</li>));

    return (
      <>
        <div>
          <ul>
            {leftList}
          </ul>
        </div>
        <div>
          <ul>
            {rightList}
          </ul>
        </div>
      </>
    );
  };


  return (
    <section className='sectionMargin'>
      <Hero />
      <Section invert={true}>
        <Card invert={true} style={{ maxWidth: '600px' }}>
          <p>
            {aboutMe.description}
          </p>
          <div style={style.listContainer}>
            {renderListOfSkills()}
          </div>
        </Card>
      </Section>
      {/* decide if I want to show the certifications or not */}
      {/* {
        isMobile
          ?
          <Section invert={false}>
            <Carousel slides={renderCertifications} />
          </Section>
          :
          <MobileCertificates />
      } */}
    </section>
  );
};

export default AboutMe;