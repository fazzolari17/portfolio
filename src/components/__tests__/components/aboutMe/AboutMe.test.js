import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import AboutMe from '../../../aboutMe/AboutMe';
import renderer from 'react-test-renderer';

describe('about me page', () => {
  describe('snapshot', () => {
    it('renders correctly', () => {
      const tree = renderer.create(<AboutMe />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
  describe('aboutMe', () => {
    it('renders the page', () => {
      const { getByText } = render(<AboutMe />);

      const name = getByText(/Giuseppe fazzolari/i);
      expect(name).toBeInTheDocument();
    });
  });
});