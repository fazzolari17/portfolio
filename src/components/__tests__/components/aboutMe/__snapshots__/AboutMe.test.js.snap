// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about me page snapshot renders correctly 1`] = `
<section
  className="sectionMargin"
>
  <section
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "var(--hero-background, #395E66)",
        "borderRadius": "5px",
        "boxShadow": "-1px 0px 38px 8px rgba(0,0,0,0.51)",
        "display": "flex",
        "flexDirection": "column",
        "fontFamily": "Lexend Deca, sans-serif",
        "margin": "2rem auto",
        "maxWidth": "300px",
        "padding": "2rem",
      }
    }
  >
    <div
      style={
        Object {
          "height": "300px",
          "overflow": "hidden",
          "width": "150px",
        }
      }
    >
      <img
        alt="headshot of the page owner Giuseppe Fazzolari"
        src="Image to pull headshot.jpeg"
        style={
          Object {
            "borderRadius": "5px",
            "height": "100%",
            "objectFit": "cover",
            "objectPosition": "25% 20px",
            "transform": "scale(1.85)",
            "width": "100%",
          }
        }
      />
    </div>
    <h1
      style={
        Object {
          "color": "var(--header-text, #F0F8FF)",
          "marginTop": "1rem",
        }
      }
    >
      Giuseppe Fazzolari
    </h1>
    <p
      style={
        Object {
          "color": "var(--hero-secondaryText, #C1AE7C)",
          "fontFamily": "Ubuntu sans-serif",
        }
      }
    >
      Software Engineer
    </p>
  </section>
  <section
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "var(--header-background, #071013)",
        "color": "var(--heaader-text, #F0F8FF)",
        "display": "flex",
        "justifyContent": "center",
        "padding": "1rem",
        "width": "100vw",
      }
    }
  >
    <section
      style={
        Object {
          "backgroundColor": "var(--header-contrastColor, #75ABBC)",
          "borderRadius": "5px",
          "boxShadow": "",
          "color": "var(--header-background, #071013)",
          "fontFamily": "Lexend Deca, sans-serif",
          "margin": "2rem auto",
          "maxWidth": "600px",
          "padding": "2rem",
        }
      }
    >
      <p>
        I'm an avid hiker, biker, backcountry camper, and hanger (Hammocking). I Love all things outdoors! I have a passion for learning, and love to be challenged. I am happiest when I am creating something. My current experience includes REST API, MERN, CRUD, CI/CD, TDD. 
      </p>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginTop": "1rem",
          }
        }
      >
        <div>
          <ul>
            <li>
              TypeScript
            </li>
            <li>
              JavaScript
            </li>
            <li>
              Git
            </li>
            <li>
              React
            </li>
            <li>
              Node
            </li>
            <li>
              React Native
            </li>
            <li>
              Mongo DB
            </li>
          </ul>
        </div>
        <div>
          <ul>
            <li>
              SQL
            </li>
            <li>
              HTML5
            </li>
            <li>
              CSS3
            </li>
            <li>
              Jest
            </li>
            <li>
              Mocha
            </li>
            <li>
              Cypress E2E
            </li>
          </ul>
        </div>
      </div>
    </section>
  </section>
  <section
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "width": "100vw",
      }
    }
  >
    <section>
      <div
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#395E66",
            "borderRadius": "5px",
            "boxShadow": "-1px 0px 38px 8px rgba(0,0,0,0.51)",
            "display": "flex",
            "flexDirection": "row",
            "fontFamily": "Lexend Deca, sans-serif",
            "height": "auto",
            "justifyContent": "center",
            "margin": "2rem auto",
            "maxHeight": "auto",
            "maxWidth": "75%",
            "padding": "2rem",
            "width": "90%",
          }
        }
      >
        <svg
          color="white"
          fill="currentColor"
          height={75}
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
              "cursor": "pointer",
            }
          }
          viewBox="0 0 16 16"
          width={75}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 0 0 16A8 8 0 0 0 8 0zm3.5 7.5a.5.5 0 0 1 0 1H5.707l2.147 2.146a.5.5 0 0 1-.708.708l-3-3a.5.5 0 0 1 0-.708l3-3a.5.5 0 1 1 .708.708L5.707 7.5H11.5z"
          />
        </svg>
        <div
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <h2
            style={
              Object {
                "color": "var(--header-text, #F0F8FF)",
                "margin": ".5rem 0 1rem 0",
              }
            }
          >
            CERTIFICATIONS
          </h2>
          <a
            href="https://www.freecodecamp.org/certification/fazzolari17/responsive-web-design"
            rel="noreferrer"
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
            target="_blank"
          >
            <img
              alt="image of responsive web design certification from FreeCodeCamp"
              src="responsiveWebDesign.png"
              style={
                Object {
                  "height": "auto",
                  "width": "70%",
                }
              }
            />
          </a>
        </div>
        <svg
          color="white"
          fill="currentColor"
          height={75}
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": "white",
              "cursor": "pointer",
            }
          }
          viewBox="0 0 16 16"
          width={75}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
          />
        </svg>
      </div>
    </section>
  </section>
</section>
`;
